crates/sim2h/src/job/connection.rs

```diff
@@ -73,22 +73,24 @@ impl ConnectionJob {
             }
             Err(crossbeam_channel::TryRecvError::Empty) => (),
         }
-        match self.wss.read(self.frame.as_mut().unwrap()) {
-            Ok(_) => {
-                trace!(
-                    "run_result: read from {} in connection",
-                    self.wss.remote_url()
-                );
-                let frame = self.frame.take().unwrap();
-                trace!("frame read from {} {:?}", self.wss.remote_url(), frame);
-                self.report_msg(Ok(frame));
-                // we got data this time, check again right away
-                return Ok(JobResult::default());
-            }
-            Err(e) if e.would_block() => (),
-            Err(e) => {
-                error!("error in read for {}: {:?}", self.wss.remote_url(), e,);
-                return Err(e.into());
+        {
+            match self.wss.read(self.frame.as_mut().unwrap()) {
+                Ok(_) => {
+                    trace!(
+                        "run_result: read from {} in connection",
+                        self.wss.remote_url()
+                    );
+                    let frame = self.frame.take().unwrap();
```

Comment:
I know this isn't part of your PR but why do we have to `unwrap()` the `take()`?

Reply:
the take api returns an option, because the value could already have been taken, and so it'll be `None` -- since we checked above if it was None add an instance to it, we can unwrap safely here